### ISSUE TITLE
Crunchyroll 404 (due to forced US beta?)

### ISSUE THREAD

**Issue by quixadhal** [yt-dlp/yt-dlp#3121]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Description

Attempting to download a single video from a season, 'https://beta.crunchyroll.com/watch/GYK53DMPR/wicked-lord-shingan-reborn' results in a 404 Not Found error, but works as expected in a browser.

I suspect this is because ALL people in the US are forcibly redirected to the "beta" site now, which is no longer really beta, and so if the extractor still tries to point at the old site, it fails for anyone who can be detected as coming from the US.  I don't currently have a way to test this via a VPN, but that would probably be the way to verify this issue.  This is not a premium video.

Since Crunchyroll now owns Funimation and has decided to remove their content from competing websites, this will probably become a larger issue going forward.

### Verbose log

```shell
[debug] Command-line config: ['-vU', '-sF', 'https://beta.crunchyroll.com/watch/GYK53DMPR/wicked-lord-shingan-reborn']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (zip)
[debug] Python version 3.9.10 (CPython 64bit) - Linux-5.16.0-4-amd64-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4.1-3 (setts), ffprobe 4.4.1-3
[debug] Optional libraries: brotli, Cryptodome, sqlite, websockets
[debug] Proxy map: {}
Latest version: 2022.03.08.1, Current version: 2022.03.08.1
yt-dlp is up to date (2022.03.08.1)
[debug] [crunchyroll:beta] Extracting URL: https://beta.crunchyroll.com/watch/GYK53DMPR/wicked-lord-shingan-reborn
[crunchyroll:beta] wicked-lord-shingan-reborn: Downloading webpage
[crunchyroll] None: Downloading webpage
[debug] [crunchyroll] Extracting URL: https://www.crunchyroll.com/love-chunibyo--other-delusions---heart-throb--/wicked-lord-shingan-reborn-648781
[crunchyroll] 648781: Downloading webpage
ERROR: [crunchyroll] 648781: Unable to download webpage: HTTP Error 404: Not Found (caused by <HTTPError 404: 'Not Found'>); please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/wiley/bin/yt-dlp/yt_dlp/extractor/common.py", line 637, in extract
    ie_result = self._real_extract(url)
  File "/home/wiley/bin/yt-dlp/yt_dlp/extractor/crunchyroll.py", line 406, in _real_extract
    webpage = self._download_webpage(
  File "/home/wiley/bin/yt-dlp/yt_dlp/extractor/crunchyroll.py", line 259, in _download_webpage
    return super(CrunchyrollBaseIE, self)._download_webpage(request, *args, **kwargs)
  File "/home/wiley/bin/yt-dlp/yt_dlp/extractor/common.py", line 928, in _download_webpage
    res = self._download_webpage_handle(
  File "/home/wiley/bin/yt-dlp/yt_dlp/extractor/common.py", line 796, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "/home/wiley/bin/yt-dlp/yt_dlp/extractor/common.py", line 781, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/home/wiley/bin/yt-dlp/yt_dlp/extractor/common.py", line 763, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/home/wiley/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3588, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.9/urllib/request.py", line 555, in error
    result = self._call_chain(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 747, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```

**Comment by qndydyhm** [comment]
NOT ALL people in the US are forcibly redirected to the "beta" site. I think it is because you agreed "[trybeta](https://www.crunchyroll.com/trybeta)", you can get the old link by clearing all cookies or quitting beta trial.

**Comment by quixadhal** [comment]
> NOT ALL people in the US are forcibly redirected to the "beta" site. I think it is because you agreed "[trybeta](https://www.crunchyroll.com/trybeta)", you can get the old link by clearing all cookies or quitting beta trial.

Well, that's certainly not true at this point in time.  There is no way to quit the beta trial anymore, which is obvious if you actually try to do so.  However, it's a moot point since yt-dlp isn't using your browser, and won't use your account unless you give it those login credentials.

Even if it were true though, do you think asking people to break their potentially paid account setup is a valid way to fix a problem?  The whole point of tools like this is to allow you to capture content you already have legitimate access to, so you don't feel the need to go to less legitimate alternative sources, no?

**Comment by qndydyhm** [comment]
> Well, that's certainly not true at this point in time. There is no way to quit the beta trial anymore, which is obvious if you actually try to do so. However, it's a moot point since yt-dlp isn't using your browser, and won't use your account unless you give it those login credentials.
> 
> Even if it were true though, do you think asking people to break their potentially paid account setup is a valid way to fix a problem? The whole point of tools like this is to allow you to capture content you already have legitimate access to, so you don't feel the need to go to less legitimate alternative sources, no?

I am not in the beta trial so I don't know there is no way to quit it. In fact,[ it](https://beta.crunchyroll.com/watch/GYK53DMPR/wicked-lord-shingan-reborn) will redirect me back to the original website [homepage](https://www.crunchyroll.com/), so yt-dlp don't work is because the beta version require the account currently in a beta trial(as you mentioned, yt-dlp is not using cookies in browser). I believe the same thing happened with the premium videos even in non-beta version since it also require an account with certain permission. 

I don't get the point of "breaking potentially paid account setup", as you mentioned it is not a premium video, then you should be able to find the non-beta url of it easily, and you don't need to find "less legitimate alternative sources".

**Comment by quixadhal** [comment]
Besides not being able to download any premium content which you may have paid for (using a login), it's also incredibly annoying to have to bookmark everything twice.  The URL format is different between the two sites.

For example, the same episode of the same show:
    https://beta.crunchyroll.com/watch/GYK53DMPR/wicked-lord-shingan-reborn
vs.
    https://www.crunchyroll.com/love-chunibyo-other-delusions-heart-throb-/episode-1-wicked-lord-shingan-reborn-648781

So, if you have an account you're already using (paid or not), and are in the US and have no choice about using the beta other than not logging in, you now can't just use your existing watch lists to populate the urls for yt-dlp, but have to go search for them and fetch them by hand from the not-logged-in version of the site.

And of course, this is just a temporary workaround.  When have you EVER known a large company to back-pedal on making their "beta" the only version available?  It's not a question of if, but when.  The US customers already have that choice taken away if they log in.  At some point, they'll promote that to all users, and then the non-logged-in version will match as well. 

However, I am not here to argue.  I reported what I see as a legitimate bug that will affect more and more yt-dlp users over time.  I clarified why it's an issue.  You are free to disagree, but we should take this to private emails since this is a bug reporting site, not a discussion forum.

**Comment by wamasi** [comment]
Try compiling the draft fix mentioned in #2930. I'm also stuck in Beta site and have been using this draft fix for a few weeks now.

**Comment by tejing1** [comment]
My PR won't fix this. It's an error with how the non-beta url is being calculated from the beta page info during the redirect. For now, just get the non-beta url directly from the website when downloading from this series. Or supply login credentials if you want to actually use the beta extractor; without them it just redirects to non-beta anyway.

In this case, the correct non-beta url would be `https://www.crunchyroll.com/love-chunibyo-other-delusions-heart-throb-/episode-1-wicked-lord-shingan-reborn-648781`

I guess the series portion of the non-beta url isn't always equal to the `series_slug_title` in the beta api. Based on this example, I'd guess it needs to squash down multiple `-`s in a row into a single `-`, but I'd like some more data before making that change.

**Comment by tejing1** [comment]
I've included a fix in #2955
